x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/SearchApplicationTemplateService.java

```diff
@@ -38,13 +43,20 @@ public SearchApplicationTemplateService(ScriptService scriptService, NamedXConte
     }
 
     public SearchSourceBuilder renderQuery(SearchApplication searchApplication, Map<String, Object> templateParams) throws IOException,
-        ValidationException {
+        ValidationException, XContentParseException {
         final SearchApplicationTemplate template = searchApplication.searchApplicationTemplateOrDefault();
         template.validateTemplateParams(templateParams);
         final Map<String, Object> renderedTemplateParams = renderTemplateParams(template, templateParams);
         final Script script = template.script();
         TemplateScript compiledTemplate = scriptService.compile(script, TemplateScript.CONTEXT).newInstance(renderedTemplateParams);
-        final String requestSource = SearchTemplateHelper.stripTrailingComma(compiledTemplate.execute());
+        String requestSource;
+        try {
+            requestSource = SearchTemplateHelper.stripTrailingComma(compiledTemplate.execute());
+        } catch (JsonProcessingException e) {
+            JsonLocation loc = e.getLocation();
+            throw new XContentParseException(new XContentLocation(loc.getLineNr(), loc.getColumnNr()), e.getMessage(), e);
```

Comment:
How friendly is this stack trace to debugging? Should we add a nicer message here?

Reply:
For background, I used this specific exception & message because it matches what the [search template API](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-template-api.html) emits in the same scenario. One of my goals with this change is to more closely align the error handling behavior of search applications & "generic" search templates to avoid more bugs like issue #101010.